src/toil/jobStores/fileJobStore.py

```diff
@@ -162,24 +164,29 @@ def jobs(self):
     # Functions that deal with temporary files associated with jobs
     ##########################################
 
+    def _copyOrLink(self, url, path):
+        # linking is not done be default because of issue #1755
+        if self.linkImports:
+            try:
+                os.link(os.path.realpath(self._extractPathFromUrl(url)), path)
```

Comment:
I would explicitly make the file read-only at this point. This will solve 2 things
1) Prevent any unexpected modification of the file before it is pulled in by readGlobalFile
2) If the file(inode) cannot be made read-only It will crash the run much earlier than if we were to wait for it to be pulled down by readGlobalFile for the first time.